olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationCredentials.h

```diff
@@ -59,6 +60,38 @@ class AUTHENTICATION_API AuthenticationCredentials {
    */
   const std::string& GetSecret() const;
 
+  /*
+   * @brief Reads credentials to access OLP from an input stream that is
+   * interpreted as a sequence of characters.
+   *
+   * The stream must contain the following sequences of characters:
+   * * here.access.key.id = <your access key ID>
+   * * here.access.key.secret = <your access key secret>
+   *
+   * @param[in] stream Stream to read the credentials from.
+   * @return Optional value with the credentials if the credentials were read
+   * successfully, or no value in case of failure.
+   */
+  static boost::optional<AuthenticationCredentials> ReadFromStream(
+      std::istream& stream);
+
+  /*
+   * @brief Parses the credentials file downloaded from the OLP website and
```

Comment:
There are two different files with credentials: `credentials.properties`  that contains platform credentials and `settings.xml` that contains repository credentials. For more information,  see [the related section](https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) in the Terms and Permissions User Guide.
I agree that users may change the file names, but we differentiate the files by their original names. Users know the original file names as they download the files. However, we cannot guess how our users may rename them :) 
Therefore, I suggest that we write the following sentence: `Parses the **credentials.properties** file downloaded from the OLP [website] (https://developer.here.com/olp/documentation/access-control/user-guide/topics/get-credentials.html) and retrieves a value with your credentials.`

Reply:
Done.